muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/workspace-journal.js

```diff
@@ -1,4 +1,76 @@
 (function() {
+  'use strict';
+
+  $.widget("custom.journal", {
+    _create : function() {
+      this._student = undefined;
+      this.loadPage(this.options.workspaceId, this._student, 1);
+
+      $(document).on('change', '#studentSelectField', $.proxy(this._onStudentSelectFieldChange, this));
+      this.element.on('click', '.wj-page-link-load-more:not(.disabled)', $.proxy(this._onLoadMoreClick, this));
+    },
+    
+    loadPage: function(workspaceId, userEntityId, page) {
+      var firstResult = (page - 1) * this.options.pageSize;
+      this._page = page;
+      
+      var params = { 
+        firstResult: firstResult, 
+        maxResults: this.options.pageSize 
+      };
+      
+      if (userEntityId)
+        params["userEntityId"] = userEntityId;
+      
+      mApi().workspace.workspaces.journal.read(workspaceId, params).callback(
+        $.proxy(function(err, journalEntries) {
+          if (err) {
+            $('.notification-queue').notificationQueue('notification', 'error', err);
+          } else {
+            var template = this.options.canListAllEntries ? 'journal/journalentries_teacher.dust' : 'journal/journalentries_my.dust';
+            
+            renderDustTemplate(template, journalEntries, function(text) {
+              $("#journalEntries").append(text);
+            });
+            
+            if (journalEntries && (journalEntries.length < this.options.pageSize))
```

Comment:
Hardly an issue but instead of disabling the button, could it simply be hidden if there are no more pages to load?